image_manager/api/views.py

```diff
@@ -0,0 +1,49 @@
+from urllib.parse import unquote
+
+from django.http import HttpResponse
+from rest_framework import viewsets, permissions
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from image_manager.api.serializer import ImageSerializer
+from image_manager.models import Image
+
+
+class ImageViewSet(viewsets.ReadOnlyModelViewSet):
+    queryset = Image.objects.all()
+    serializer_class = ImageSerializer
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get_queryset(self):
+        return self.queryset.filter(uploader=self.request.user)
+
+    @action(methods=['POST'], detail=False)
+    def upload(self, request):
+        image_urls = request.data.get('urls', [])
+
+        for image_url in image_urls:
+            image_url = unquote(image_url)
+
+            if Image.objects.filter(url=image_url).exists():
+                image = Image.objects.get(url=image_url)
+            else:
+                image = Image.objects.create(url=image_url, uploader=request.user)
+
+            image.download()
```

Comment:
The current implementation of `upload` is synchronous. If the list of URLs is long, this could lead to long processing times for the request. 

Using a task queue like Celery could be used to enable asynchronous processing.